inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationScopeResolver.java

```diff
@@ -0,0 +1,257 @@
+        } else if (arguments.length == 0) {
+            return takesArguments(0);
+        } else {
+            MatcherBuilder<MethodDescription> builder = new MatcherBuilder<>();
+
+            for (int i = 0; i < arguments.length; i++) {
+                builder.and(takesArgument(i, named(arguments[i])));
+            }
+
+            return builder.build();
+        }
+    }
+
+    /**
+     * Creating an {@link ElementMatcher} matching methods with the given access modifiers.
+     */
+    private ElementMatcher.Junction<MethodDescription> buildVisibilityMatcher(AccessModifier[] modifiers) {
+        if (modifiers == null) {
+            return null;
+        }
+
+        List<AccessModifier> accessModifiers = Arrays.asList(modifiers);
+        boolean isPublic = accessModifiers.contains(AccessModifier.PUBLIC);
+        boolean isProtected = accessModifiers.contains(AccessModifier.PROTECTED);
+        boolean isPackage = accessModifiers.contains(AccessModifier.PACKAGE);
+        boolean isPrivate = accessModifiers.contains(AccessModifier.PRIVATE);
+        if (isPublic && isProtected && isPackage && isPrivate) {
+            return null;
+        }
+
+        MatcherBuilder<MethodDescription> builder = new MatcherBuilder<>();
+
+        for (AccessModifier modifier : modifiers) {
+            switch (modifier) {
+                case PUBLIC:
+                    builder.or(isPublic());
+                    break;
+                case PROTECTED:
+                    builder.or(isProtected());
+                    break;
+                case PACKAGE:
+                    builder.or(isPackagePrivate());
+                    break;
+                case PRIVATE:
+                    builder.or(isPrivate());
+                    break;
+            }
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Creates an {@link ElementMatcher} matching items with the given name settings.
+     */
+    private <N extends NamedElement> ElementMatcher.Junction<N> buildNameMatcher(NameMatcherSettings matcherSettings) {
+        String namePattern = matcherSettings.getNamePattern();
+
+        if (StringUtils.isNotEmpty(namePattern)) {
+            StringMatcher.Mode matcherMode = matcherSettings.getMatcherMode();
+            return new NameMatcher<>(new StringMatcher(namePattern, matcherMode));
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Creates the {@link ElementMatcher} which is matching the scope's type specified in the configuration.
+     */
+    private ElementMatcher.Junction<TypeDescription> buildTypeMatcher(InstrumentationScopeSettings scopeSettings) {
+        if (scopeSettings.getTypeScope() == null) {
+            return null;
+        }
+
+        MatcherBuilder<TypeDescription> builder = new MatcherBuilder<>();
+
+        if (scopeSettings.getTypeScope().getSuperclass() != null) {
+            processSuperType(builder, scopeSettings.getTypeScope().getSuperclass());

```

Comment:
For this matcher you need an additional check to make sure it is a class.
And for the interfaces you need a check to ensure it is an interface. Otherwise you could match for interfaces on the superclass config and for classes on the interface config.

Reply:
Done.